src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -296,9 +303,13 @@ namespace winrt::TerminalApp::implementation
     //   of a command.
     // - Only one dialog can be visible at a time. If another dialog is visible
     //   when this is called, nothing happens. See _ShowDialog for details
-    void TerminalPage::_ShowMultiLinePasteWarningDialog()
+    winrt::Windows::Foundation::IAsyncOperation<ContentDialogResult> TerminalPage::_ShowMultiLinePasteWarningDialog()
     {
-        _showDialogHandlers(*this, FindName(L"MultiLinePasteDialog").try_as<WUX::Controls::ContentDialog>());
+        if (auto presenter{ _dialogPresenter.get() })
+        {
+            co_return co_await presenter.ShowDialog(FindName(L"MultiLinePasteDialog").try_as<WUX::Controls::ContentDialog>());
```

Comment:
`co_return co_await` definitely sounds like a `return`

Unless coroutines do not work the way I think they do.

This should be "return somebody else's async as my async", right?

Reply:
Same as above. If I remove the `co_await`, then this would return a `IAsyncOperation<IAsyncOperation<ContentDialogResult>>`.

Reply:
Nah, this is right.. it's just really weird to me that it works this way.

Because `presenter.ShowDialog` returns an `AsyncAction<T>`, I would expect:

* `co_await` would return `T`
* `co_return` would capture that `T` and return `AsyncAction<T>`
* therefore, `return` without any `co_` would return _the literal return value of `ShowDialog`_, which ... would be `AsyncAction<T>`

I'm surprised (in the direction of the C++ standard committee) that that isn't the case.

Reply:
But I guess,

> Coroutines cannot use variadic arguments, **plain return statements**

emphasis mine.

I'll mark these as Resolved.